workflow-templates/test-package.yaml

```diff
@@ -7,14 +7,17 @@ on:
   pull_request:
     branches: [ $default-branch ]
   schedule:
+    # “At 00:00 (UTC) on Sunday.”
     - cron: "0 0 * * 0"
 
 env:
   PUB_ENVIRONMENT: bot.github
 
+permissions: read-all
+
 jobs:
-  # Check code formatting and static analysis on a single OS (linux)
-  # against Dart dev and stable.
+  # Check code formatting and static analysis on a single OS (linux) against
+  # Dart dev and stable.
   analyze:
     runs-on: ubuntu-latest
     strategy:
```

Comment:
The version pins in this workflow should use hashes rather than tags.

Reply:
I'll update them. I was concerned with these getting out of date - of them not being updated by dependabot.